cni/pkg/nodeagent/cni-watcher.go

```diff
@@ -200,3 +186,32 @@ func (s *CniPluginServer) ReconcileCNIAddEvent(ctx context.Context, addCmd CNIPl
 
 	return nil
 }
+
+func (s *CniPluginServer) getPodWithRetry(log *istiolog.Scope, name, namespace string) (*corev1.Pod, error) {
```

Comment:
I think this was mentioned in other PRs, but if this is now its own function, it would be good to cover behavior with unit tests in `cni-watcher_test.go` (the lack of this  is probably why we had this bug in the first place)

We didn't do that before because it would be a slow test due to the timeouts, but if this is a private func, can we just pass the timeouts in as args and test quick iterations in a unit test to codify correct behavior here?

Reply:
The way we have things abstracted actually still makes this pretty hard since we don't actually call this except in cni-watcher where all them mocking makes i tricky to flow through

Reply:
Looks like it could be a copypaste of one of the existing tests (e.g. `TestCNIPluginServer`), but we simply call `ReconcileCNIAddEvent` without adding an underlying pod to the fake client, right?